### PR TITLE
Python: Add support for static table

### PR DESCRIPTION
**Pull request by rubenvdg**
This PR proposes adding support for static tables (i.e., reading a table directly from a metadata file without using a catalog, see also https://github.com/apache/iceberg/issues/6430). Happy to hear if this makes sense.

Regarding unit tests:
We could run all the existing `Table` unit tests for `StaticTable` too, but that might be a bit artificial. You'd get something ugly like this:

```python
@pytest.fixture
def table(..) -> Table:
    ...

@pytest.fixture
def static_table(...) -> StaticTable:
    ...

@pytest.mark.parametrize("the_table", ["table", "static_table"])
def test_schema(the_table: Table, request: FixtureRequest) -> None:
    assert request.getfixturevalue(the_table).schema() == Schema(...)
```

resolves https://github.com/apache/iceberg/issues/6430

### CODE REVIEW

File: python/pyiceberg/exceptions.py [hunk 1]
```diff
@@ -90,3 +90,7 @@ class SignError(Exception):
 
 class ResolveError(Exception):
     pass
+
+
+class StaticTableImmutableError(Exception):
```
Comment (Fokko) [inline python/pyiceberg/exceptions.py:95]: How do you feel like re-using the internal `NotImplementedError`?

File: python/pyiceberg/table/__init__.py [hunk 2]
```diff
@@ -167,6 +169,32 @@ def __eq__(self, other: Any) -> bool:
...
+
+        return cls(
+            identifier=("static-table", metadata_location),
+            metadata_location=metadata_location,
+            metadata=metadata,
+            io=load_file_io({**properties, **metadata.properties}),
+        )
+
+    @staticmethod
+    def _load_metadata(metadata_location: str, properties: Properties) -> TableMetadata:
```
Comment (Fokko) [inline python/pyiceberg/table/__init__.py:192]: How do you feel about merging this logic into `from_metadata`? I t looks like we don't reuse this anywhere.

File: python/pyiceberg/catalog/rest.py [hunk 3]
```diff
@@ -175,11 +175,7 @@ class RestCatalog(Catalog):
     session: Session
     properties: Properties
 
-    def __init__(
-        self,
-        name: str,
-        **properties: str,
-    ):
+    def __init__(self, name: str, **properties: str):
```
Comment (Fokko) [inline python/pyiceberg/catalog/rest.py:178]: Why did this change? I always assumed that black was deterministic.

**Review by Fokko — COMMENTED** [review]
@rubenvdg I think this is great! 👏🏻 Few small comments and one important thing is missing: docs! 📖 Could you add some under `python/mkdocs/docs/api.md` on how to load a StaticTable? Regarding the tests, I think that's okay to not duplicate all the tests.

File: python/pyiceberg/catalog/rest.py (hunk 3 above)
Comment (rubenvdg) [reply python/pyiceberg/catalog/rest.py:178]: I removed the trailing comma

File: python/pyiceberg/exceptions.py (hunk 1 above)
Comment (rubenvdg) [reply python/pyiceberg/exceptions.py:95]: My reasoning was that for a `StaticTable` refreshing metadata doesn't make sense, hence it's never going to be implemented, hence `NotImplementedError` doesn't make sense. 

I might have misunderstood the whole purpose of `refresh()`.

**Comment by rubenvdg** [comment]
Added docs and fixed the rest.

File: python/pyiceberg/catalog/rest.py (hunk 3 above)
Comment (Fokko) [reply python/pyiceberg/catalog/rest.py:178]: Nice!

**Review by Fokko — APPROVED** [review]
Thanks for working on this @rubenvdg Looks good to me!

**Comment by Fokko** [comment]
@rubenvdg can you rebase? We were waiting for https://github.com/apache/iceberg/pull/6719 to be merged so we don't update the docs right away

**Comment by rubenvdg** [comment]
Tuuuuuuurlijk maestro

**Comment by Fokko** [comment]
Thanks again @rubenvdg for picking this up 👏🏻

**Pull request merged**